openhands/integrations/utils.py

```diff
@@ -12,35 +11,52 @@ async def validate_provider_token(
     token: SecretStr, base_domain: str | None = None
 ) -> ProviderType | None:
     """
-    Determine whether a token is for GitHub or GitLab by attempting to get user info
-    from both services.
+    Determine whether a token is for GitHub, GitLab, or Bitbucket by attempting to get user info
+    from the services.
 
     Args:
         token: The token to check
+        base_domain: Optional base domain for the service
 
     Returns:
         'github' if it's a GitHub token
         'gitlab' if it's a GitLab token
-        None if the token is invalid for both services
+        'bitbucket' if it's a Bitbucket token
+        None if the token is invalid for all services
     """
+    # Skip validation for empty tokens
+    if token is None or not token.get_secret_value().strip():
```

Comment:
```suggestion
    if token is None:
```

Pydantic will check for the case its `SecretStr('')` so we don't need to get the secret value here